test/mongodb.test.js

```diff
@@ -1072,6 +1072,111 @@ describe('mongodb connector', function () {
     });
   });
 
+  describe('replaceOrCreate', function() {
+    it('should create a model instance even if it already exists', function(done) {
+      Product.replaceOrCreate({name: 'newFoo'}, function(err, updatedProduct) {
+        should.not.exist(err);
```

Comment:
This produces unhelpful error messages. Please always use `if (err) return done(err)` in unit-tests, even if that's not consistent with the legacy code around.